lib/charms/layer/apache_bigtop_base.py

```diff
@@ -98,13 +100,35 @@ def install(self):
         You will then need to call `render_site_yaml` to set up the correct
         configuration and `trigger_puppet` to install the desired components.
         """
+        self.install_java()
         self.pin_bigtop_packages()
         self.check_reverse_dns()
         self.fetch_bigtop_release()
         self.install_puppet_modules()
         self.apply_patches()
         self.render_hiera_yaml()
 
+    def install_java(self):
+        """
+        Possibly install java.
+
+        """
+        java_package = self.options.get("install_java")
+        if not java_package:
+            # noop if we are setting up the openjdk relation.
+            return
+
+        _, _, series = platform.dist()
```

Comment:
Unfortunately, this is deprecated in Python 3.5.  Probably not an issue any time soon, but it might be more proactive to use `charmhelpers.host.lsb_release()['DISTRIB_CODENAME']` instead.